scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,384 @@
+        ftpClient.connect(ftpHostname, ftpPort);
+        int reply = ftpClient.getReplyCode();
+        if (!FTPReply.isPositiveCompletion(reply)) {
+            ftpClient.disconnect();
+            throw new IOException("Exception in connecting to FTP Server");
+        }
+        ftpClient.login(ftpUsername, ftpPassword);
+
+        // use local passive mode to pass firewall
+        ftpClient.enterLocalPassiveMode();
+
+        getOut().println("Connected");
+
+        switch (ftpMode) {
+            //FTP mode is GET
+            case "GET":
+                filesRelativePathName = FtpGet(ftpClient);
+                break;
+
+            //FTP mode is PUT
+            case "PUT":
+                filesRelativePathName = FtpPut(ftpClient);
+                break;
+
+            default:
+                throw new RuntimeException("wrong value");
+        }
+        // log out and disconnect from the server
+        ftpClient.logout();
+        ftpClient.disconnect();
+
+        getOut().println("Disconnected");
+        return (Serializable) filesRelativePathName;
+    }
+
+    private List<String> FtpGet(FTPClient ftpClient) throws IOException {
+        List<String> filesRelativePathName;
+        getOut().println("Importing file(s) from " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+        FTPFile[] ftpFile = ftpClient.listFiles(ftpRemoteFilePath);
+        if (ftpFile.length == 0) {
+            throw new IllegalArgumentException(ftpRemoteFilePath + " not found. Please, enter a valid path.");
+        }
+        filesRelativePathName = new ArrayList<>();
+        // If it is a single file:
+        if (ftpFile.length == 1 && ftpRemoteFilePath.contains(ftpFile[0].getName())) {
+            String saveFilePath = Paths.get(ftpLocalFilePath, ftpFile[0].getName()).toString();
+            if (downloadSingleFile(ftpClient, ftpRemoteFilePath, saveFilePath)) {
+                getOut().println("Succeed to download the file " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+                filesRelativePathName.add(saveFilePath);
+            } else {
+                throw new IOException("Error: COULD NOT download the file: " + ftpRemoteFilePath);
+            }
+
+            // If the file is a zip, and ftpExtractArchive is set to true
+            if (ftpExtractArchive && ftpRemoteFilePath.endsWith(".zip")) {
+                ZipUtil.unpack(new File(saveFilePath), new File(ftpLocalFilePath));
+                //extractArchive(saveFilePath);
+            }
+        } // If it is a folder, download all its contents recursively
+        else {
+            ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+            filesRelativePathName.addAll(new HashSet(downloadDirectory(ftpClient, "", "", ftpLocalFilePath)));
+        }
+        getOut().println("END Import file(s) from FTP.");
+        return filesRelativePathName;
+    }
+
+    private List<String> FtpPut(FTPClient ftpClient) throws IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)